### PR TITLE
Add retrofit-clients project

### PR DESCRIPTION
**Pull request by uschi2000**
@pnepywoda

### CODE REVIEW

File: retrofit-clients/src/main/java/com/palantir/remoting/retrofit/RetrofitClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        okClient.interceptors().add(retryInterceptor);
+
+        // ssl
+        okClient.setSslSocketFactory(sslSocketFactory.orNull());
+
+        return new OkClient(okClient);
+    }
+
+    public static <T> T createProxy(Optional<SSLSocketFactory> sslSocketFactoryOptional, String uri, Class<T> type,
+            OkHttpClientOptions options) {
```
Comment (markelliot) [inline retrofit-clients/src/main/java/com/palantir/remoting/retrofit/RetrofitClientFactory.java:59]: We don't have a failover variant of this?

Comment (nmiyake) [inline retrofit-clients/src/main/java/com/palantir/remoting/retrofit/RetrofitClientFactory.java:59]: I wrote a Retrofit fail-over variant for one of my projects. Feel free to grab/take from that if you'd like.

Comment (nmiyake) [inline retrofit-clients/src/main/java/com/palantir/remoting/retrofit/RetrofitClientFactory.java]: (Core code for this is at https://github.palantir.build/nmiyake/retrofit-failover/blob/develop/retrofit-failover/src/main/java/com/palantir/retrofit/failover/FailoverInterceptor.java)

**Comment by markelliot** [comment]
Going to merge this. We should grab Nick's code in a later commit.

**Comment by uschi2000** [comment]
Cool. Will make a new release of this together with the feign 8.15.0 update.

File: retrofit-clients/src/main/java/com/palantir/remoting/retrofit/RetrofitClientFactory.java (hunk 1 above)
Comment (ash211) [inline retrofit-clients/src/main/java/com/palantir/remoting/retrofit/RetrofitClientFactory.java]: @nmiyake http-remoting is a public project and those links won't work for any non-Palantirians who see them in the future.

Comment (nmiyake) [inline retrofit-clients/src/main/java/com/palantir/remoting/retrofit/RetrofitClientFactory.java:59]: Oh yeah good call -- updated/removed comments (easy to forget GH versus GHE when working on a lot of different projects concurrently)

**Pull request merged**